### ISSUE TITLE
Intermitten error concerning "path" argument 

### ISSUE THREAD

**Issue by Orionsg** [chriscamicas/vscode-abl#37]
I have started to regularly receivethe following error when creating or opening ABL files in VS Code:
-------------------------------
Error: TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type undefined.

Source: OpenEdge ABL (Extension)
-------------------------------

Unfortunately, it is an intermittent error, i.e. sometimes is occurs many times, then not at all for a period. So I have no particular way of reproducing it.

**Comment by chriscamicas** [comment]
It's a tricky one ...
I never experienced such an error.
Does it happen even if the VScode instance has been running for a long time, and has already opened other ABL files ?
I'm trying to see if it's an extension loading issue (ie file open while extension not ready yet), or something else.
thanks

**Comment by Orionsg** [comment]
@chriscamicas 

It seems to happen when editing files, whether new ones or older ones. I am also using the VIM extension so I noticed it when going into insert mode and changing the text. But disabling the VIM extension and restarting VScode makes no difference, once I start editing and moving around the file, the problem occurs. It does not make a difference if VScode has been running for some time with several ABL files opened.

**Comment by Orionsg** [comment]
The issues is quite annoying as editor constantly lags when editing the text of files. It seems there is some algorithm which is executed every time is code is changed. Since it runs into the problem there is quite a lag in the flow of text being entered, making the editing experience horrible. The pleasure of having switched to VSCode for OpenEdge files is slowly but surely being eroded because of this simple, but incessant and debilitating problem.

Is there anything you can do to debug it? 
Let me know if there is anything I can do to help you locating the cause of the problem.

**Comment by chriscamicas** [comment]
I'm really sorry this error is hurting your experience with this extension.
Have you considered cloning this repo and debugging the extension on your side ?
You can run a VSCode instance with this extension enabled while in debug mode (attached to another VSCode).
More infos available here: https://vscode-docs.readthedocs.io/en/stable/extensions/debugging-extensions/
In the meantime, I'll try to harden some code, or log more things, but not quite sure where.

**Comment by chriscamicas** [comment]
Could you share your `.openedge.json` file ?
Could you share a small non sensitive project (ie vscode workspace) where this error occurs ?
Thank you

**Comment by chriscamicas** [comment]
you can also try to disable the Automatic syntax checking on file save in your User or Workspace preference
![image](https://user-images.githubusercontent.com/7702519/61119714-bf6d2c80-a49b-11e9-849b-0cc645c39d3f.png)

**Comment by Orionsg** [comment]
I have not been using any .open.edge.json file as I just needed editing with Intellisense, color coding, etc. This seems to be the problem, I have created a simple .openedge.json file and now the editing is fast again without the error message constantly appearing.

The problem was not there some months ago. Perhaps in one of the recent versions, code was introduced that expects the .openedge.json file to always be there.

Can you check if you get the same problem if you don't have any .openedge.json file.

**Comment by chriscamicas** [comment]
I'm glad you found a workaround.
I'm not able to reproduce this error.
What I tried:
1- create a new empty folder
2- open it with vscode (openedge abl extension activated)
3- create a new file `test.p`
4- write a simple statement `message "hello".`
5- save file
6- run this file

no error so far
I also tried in an existing project by removing the `.openedge.json` file
I'm wondering if you have the DLC environment variable setup in your system ?
If so, is the value correct ?

**Comment by chriscamicas** [comment]
Ok found it, if I remove the DLC env var, I have the same error.

**Comment by chriscamicas** [comment]
As stated in the README file, the `DLC` env var is a prerequisite. 
But, for better usability I added a check at the extension activation to verify the configuration (DLC, openedge.json, and progress binary).
I also added some documentation in the wiki to explain how to setup a project.
https://github.com/chriscamicas/vscode-abl/wiki

Thanks for your patience and I hope you'll enjoy this extension.
